macaroon.js

```diff
@@ -645,57 +764,76 @@ const Macaroon = class Macaroon {
 
   /**
     Returns the V2 JSON serialization of this macaroon.
-    @return {Object} JSON serialization of this macaroon.
+    @return {Object} JSON compatible representation of this macaroon.
   */
-  serializeJson() {
+  _exportAsJSONObjectV2() {
     const obj = {
       v: 2, // version
-      i: this._identifier,
-      s64: base64url(sjcl.codec.base64.fromBits(this._signature)),
     };
-    if (this._location) {
-      obj.l = this._location;
+    setJSONFieldV2(obj, 's', bitsToBytes(this._signatureBits));
+    setJSONFieldV2(obj, 'i', bitsToBytes(this._identifierBits));
+    if (this._locationStr) {
+      obj.l = this._locationStr;
+    }
+    if (this._caveats && this._caveats.length > 0) {
+      obj.c = this._caveats.map(caveat => {
+        const caveatObj = {};
+        setJSONFieldV2(caveatObj, 'i', bitsToBytes(caveat._identifierBits));
+        if (caveat._vidBits) {
+          setJSONFieldV2(caveatObj, 'v', bitsToBytes(caveat._vidBits));
+          caveatObj.l = caveat._locationStr;
+        }
+        return caveatObj;
+      });
     }
-    obj.c = this._caveats.map(caveat => {
-      const caveatObj = {
-        i: caveat._identifier
-      };
-      if (caveat._vid) {
-        // Use URL encoding and do not append "=" characters.
-        caveatObj.v64 = sjcl.codec.base64.fromBits(caveat._vid, true, true);
-        caveatObj.l = caveat._location;
-      }
-      return caveatObj;
-    });
     return obj;
   }
 
   /**
+   * Serializes the macaroon using the v1 binary format.
+   * @return {Uint8Array} Serialized macaroon
+   */
+  _serializeBinaryV2() {
```

Comment:
I think this method name is wrong

Reply:
Indeed. I'm quite surprised that eslint didn't catch this. Fixed.